central/complianceoperator/v2/report/manager/complianceReportgenerator/report_gen_impl.go

```diff
@@ -84,88 +92,151 @@ type ResultEmail struct {
+				return errors.New("Profile not found")
+			}
+
+			row.Profile = profiles[0].GetName()
+
+			// get remediation for the check result
+			q = search.NewQueryBuilder().AddExactMatches(search.ComplianceOperatorCheckName, c.GetCheckName()).AddExactMatches(search.ClusterID, c.GetClusterId()).ProtoQuery()
+			remediations, err := rg.remediationDS.SearchRemediations(req.Ctx, q)
+			if err != nil {
+				return err
+			}
+			if len(remediations) > 0 {
+				row.Remediation = remediations[0].GetName()
+			}
+
+			resultCluster = append(resultCluster, row)
+			if c.GetStatus() == storage.ComplianceOperatorCheckResultV2_FAIL {
+				resultEmailComplianceReport.TotalFail += 1
+			} else if c.GetStatus() == storage.ComplianceOperatorCheckResultV2_PASS {
+				resultEmailComplianceReport.TotalPass += 1
+			} else {
+				resultEmailComplianceReport.TotalMixed += 1
+			}
+			return nil
+		})
+		if err != nil {
+			return nil, err
+		}
+
+		resultsCSV[clusterID] = resultCluster
+	}
+	resultEmailComplianceReport.Clusters = len(req.ClusterIDs)
+	resultEmailComplianceReport.Profiles = req.Profiles
+	resultEmailComplianceReport.ResultCSVs = resultsCSV
+	return resultEmailComplianceReport, nil
 }
 
 func (rg *complianceReportGeneratorImpl) sendEmail(ctx context.Context, zipData *bytes.Buffer, emailBody *formatBody, formatEmailSub *formatSubject, notifiersList []*storage.NotifierConfiguration) {
 
 	errorList := errorhelpers.NewErrorList("Error sending compliance report email notifications")
-	for _, notifier := range notifiersList {
-		nf := rg.notificationProcessor.GetNotifier(ctx, notifier.GetEmailConfig().GetNotifierId())
+	for _, repNotifier := range notifiersList {
+		notifierID := repNotifier.GetId()
+		nf := rg.notificationProcessor.GetNotifier(ctx, notifierID)
 		reportNotifier, ok := nf.(notifiers.ReportNotifier)
+		log.Infof("nf is %s", nf)
+		log.Infof("reportnotifier is %s", reportNotifier)
 		if !ok {
-			errorList.AddError(errors.Errorf("incorrect type of notifier '%s' for compliance report", notifier.GetEmailConfig().GetNotifierId()))
+			errorList.AddError(errors.Errorf("incorrect type of notifier %s for compliance report", repNotifier.GetEmailConfig().GetNotifierId()))
 			continue
 		}
 
-		customBody := notifier.GetEmailConfig().GetCustomBody()
-		emailBody, err := formatEmailBodywithDetails(defaultEmailBodyTemplate, emailBody)
+		customBody := repNotifier.GetEmailConfig().GetCustomBody()
+		body, err := formatEmailBodywithDetails(defaultEmailBodyTemplate, emailBody)
+		log.Infof("format email body is %s.body is %s", emailBody, body)
 		if err != nil {
-			errorList.AddError(errors.Errorf("Error sending compliance report email for notifier '%s': %s",
-				notifier.GetEmailConfig().GetNotifierId(), err))
+			errorList.AddError(errors.Errorf("Error formatting email body for notifier %s: %s",
+				repNotifier.GetEmailConfig().GetNotifierId(), err))
 		}
 		if customBody != "" {
-			emailBody = customBody
+			body = customBody
```

Comment:
Have you tested `customBody`?

Reply:
yes I have and it is in the video I posted on slack